app/src/main/java/org/thoughtcrime/securesms/conversation/v2/settings/ConversationSettingsViewModel.kt

```diff
@@ -71,11 +87,26 @@ class ConversationSettingsViewModel @AssistedInject constructor(
     }
 
     init {
+        // update data when we have a recipient and update when there are changes from the thread or recipient
         viewModelScope.launch(Dispatchers.Default) {
-            repository.recipientUpdateFlow(threadId).collect{
-                recipient = it
-                getStateFromRecipient()
-            }
+            repository.recipientUpdateFlow(threadId) // get the recipient
+                .flatMapLatest { recipient -> // get updates from the thread or recipient
+                    merge(
+                        context.contentResolver
+                            .observeQuery(DatabaseContentProviders.Recipient.CONTENT_URI), // recipient updates
+                        (context.contentResolver.observeChanges(
+                            DatabaseContentProviders.Conversation.getUriForThread(threadId)
+                        ) as Flow<*>) // thread updates
+                    ).map {
+                        recipient // return the recipient
+                    }
+                        .onStart { emit(recipient) } // make sure there's a value straight away
+                        .debounce(200L)
```

Comment:
I suppose we don't want to debounce the first item? Which mean it should go before `onStart` I assume?

```suggestion
                        .debounce(200L)
                        .onStart { emit(recipient) } // make sure there's a value straight away
```